controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -182,10 +187,47 @@ def test_delete_volume_return_volume_delete_errors(self, mock_warning):
         with self.assertRaises(CLIFailureError):
             self.svc.delete_volume("vol")
 
+    @patch("controller.array_action.array_mediator_svc.is_warning_message")
+    def test_delete_volume_has_snapshot_fcmaps_not_removed(self, mock_warning):
+        cli_volume = self._get_source_cli_vol()
+        cli_volume.FC_id = 'many'
+        self.svc.client.svcinfo.lsvdisk.return_value = Mock(as_single_element=cli_volume)
+        mock_warning.return_value = False
+        fcmaps_as_source = self.fcmaps
+        fcmaps_as_source[0].copy_rate = "0"
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=[]), Mock(as_list=fcmaps_as_source)]
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
+            self.svc.delete_volume("vol")
+
+    @patch("controller.array_action.array_mediator_svc.is_warning_message")
+    def test_delete_volume_still_copy_fcmaps_not_removed(self, mock_warning):
+        cli_volume = self._get_source_cli_vol()
+        cli_volume.FC_id = 'many'
+        self.svc.client.svcinfo.lsvdisk.return_value = Mock(as_single_element=cli_volume)
+        mock_warning.return_value = False
+        fcmaps_as_source = self.fcmaps
```

Comment:
all of these lines seem duplicated.
could you make a new `_prepare_object_still_in_use_mocks` or something?

Reply:
done